BuildAutoRest.ps1

```diff
@@ -255,7 +267,8 @@ ForEach ($SDK In $SDKName)
             # Temp workaround untill autorest updates to use Pester V5 syntax
             $testModuleContent = Get-Content -Path:($testModulePath) -Raw
             $PesterTestsContent = Get-Content -Path:($RunPesterTestsFilePath) -Raw
-            $InvokePesterLine = $testModuleContent | Select-String -Pattern 'Invoke-Pester.*?.xml"\)'
+            # Replace if else Invoke-Pester block with contents from runpestertests file. 
+            $InvokePesterLine = $testModuleContent | Select-String -Pattern '(.*?if)(.*?\(\$null -ne \$TestName\))([\s\S]*?)(Invoke-Pester.*?.xml"\))([\s\S]*?)(Invoke-Pester.*?.xml"\))([\s\S]*?)(\})'
```

Comment:
I believe this was a change required by a new version of Autorest, either way the tests were not generating correctly until this regex pattern was swapped out